### PR TITLE
Show Instabug survey after N messages sent

### LINKED ISSUE #3738: Send user events to Instabug so that surveys can be triggered
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a UXR, I want to create Instabug surveys after specific app events so that I can survey users with relevant context.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: 

UXR is tasked with creating surveys to measure user feedback and OKR results such as ">95% of 20+ people surveyed trust Status for messaging".  In order to ensure the survey responses are relevant they need to be shows to users at the right time. For example, showing a messaging reliability survey to users when they first open the app won't yield meaningful answers, but showing the same survey to users after they send a message will ensure the users have the proper messaging context.

Instabug has a feature that can show surveys to users based on events and conditions, but in order for it to work events need to be sent to Instabug. Currently no events are sent.

#### Expected behavior
[comment]: # (Describe what you expected to happen.)

UXR creates a survey and sets a conditions to show the survey after a specific event.
When the app users completes the event, the survey is shown.

#### Actual behavior
[comment]: # (Describe what actually happened.)

UXR creates a survey, but can't set conditions because no events are being sent. 
Without conditions the survey is shown to all users at the same time.

### Solution
[comment]: # (Please summarise the solution and provide a task list on what needs to be fixed.)
*Summary*: 

Status now sends (opt-in) user events to Mixpanel (see here https://github.com/status-im/status-react/pull/3605). The same events (see https://github.com/status-im/status-react/blob/develop/src/status_im/utils/mixpanel_events.edn for a list of events) can be sent to Instabug using the logging feature described here https://docs.instabug.com/docs/react-native-logging#section-user-steps.

Like Mixpanel, we should only send these events for users that opt-in.

cc @hesterbruikman @jpbowen 

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/status-im/status-react">status-im/status-react</a></strong></div><div>status-react - a free (libre) open source, mobile OS for Ethereum</div></blockquote>
<blockquote><div><strong><a href="https://docs.instabug.com/docs/react-native-logging#section-user-steps">Logging</a></strong></div><div>You can send a variety of logs types with each crash or bug report. They will appear withing each report on your Instabug dashboard as shown below.   ## Instabug Log  You can log messages throughout your application's lifecycle. Those logs are going to be sent with each report.  ## User Steps  Insta...</div></blockquote>

### PR DESCRIPTION
**Pull request by dmitryn**
Related to #3738 

Show Instabug custom surveys when user sends N messages.
Based on temporary solution until Instabug fix User Events API.

status: ready

### CODE REVIEW

**Comment by jeluard** [comment]
@dmitryn looks good!

One comment: I am a little perplex about sending all events to both platforms.

**Comment by dmitryn** [comment]
>One comment: I am a little perplex about sending all events to both platforms.

@jeluard Good point. Agree with sending all events to Mixpanel. 
But I would reduce events sent to Instabug just to those actually needed to trigger a survey, i.e. "send-current-message" in our particular case. We could then gradually add more events needed to trigger a new custom survey when needed.
@chadyj  WDYT?

**Comment by chadyj** [comment]
> But i would reduce events sent to Instabug just to those actually needed to trigger a survey, i.e. "send-current-message" in our particular case. We could then gradually add more events needed to trigger a new custom survey when needed.

In principle that is fine but in practice it means we can't decouple surveys/UXR from engineering. Every new survey needs new engineering work, discussions, PR's, testing, then waiting for the release.  With all the events UXR isn't blocked for future surveys.

In addition to surveys, the user events are helpful to debug issues in instabug as we can get context on what the user was doing for crashes, etc.

**Comment by dmitryn** [comment]
>We faced a problem with the UserEvents and UserAttributes and we are currently working on a fix for it. 

Instabug support is working on fixing `logUserEventWithNameAndParams` method

**Comment by dmitryn** [comment]
Just got response from Instabug support regarding events sent with `logUserEventWithNameAndParams` not showing up in dashboard:

>We faced a problem with the UserEvents and UserAttributes and we are currently working on a fix for it. 

Just sent couple of more dummy events using `Instabug.logUserEventWithName("OnFeedbackButtonClicked")` and `Instabug.logUserEventWithNameAndParams("OnFeedbackButtonClicked",{"key": "value"});` to test if it works.

**Comment by flexsurfer** [comment]
@dmitryn is it still relevant, or can we close it?

**Comment by dmitryn** [comment]
Still WIP. Waiting for Instabug to fix the issue.

File: src/status_im/utils/instabug.cljs [hunk 1]
```diff
@@ -3,6 +3,38 @@
...
 
+(def instabug rn-dependencies/instabug)
+
+;; `event` is an event name, e.g. "Tap"
+;; `properties` is a map of event details or nil, e.g. {:target :send-current-message}
+;; (see mixpanel_events.edn for list of trackable events)
+
+(def survey-triggers
+  [{:event :send-current-message :count 4  :token "UqtvIKgVDUTo4l_sDS-fwA"}
+   {:event :send-current-message :count 29 :token "Hr9Dk3krPK7PPxuDbHAmXg"}])
```
Comment (goranjovic) [inline src/status_im/utils/instabug.cljs:14]: What is this token? Is it ok to just leave it open in the code like this?

Comment (chadyj) [reply src/status_im/utils/instabug.cljs:14]: It is the ID of the survey to show.

So it triggers survey "Messaging UX" after 4 messages sent, and the other "Confidence Test Survey April 2018" after 29 messages?

cc @jpbowen @hesterbruikman

**Review by goranjovic — APPROVED** [review]

File: src/status_im/utils/instabug.cljs (hunk 1 above)
Comment (dmitryn) [reply src/status_im/utils/instabug.cljs:14]: @goranjovic each token corresponds to particular survey created on Instabug.

@chadyj it shows "Messaging UX" after 5 messages
and "Confidence Test Survey April 2018" after 30 messages (don't look at numbers 4 and 29 in the code, it's 5 and 30 actually)

**Comment by antdanchenko** [comment]
@dmitryn overall looks great, no issues found, yet :)

however the feature will break several automated tests, is it possible to add an ability to disable the surveys for .env.e2e?

**Comment by dmitryn** [comment]
@antdanchenko added `INSTABUG_SURVEYS=0` flag for e2e

File: src/status_im/utils/instabug.cljs (hunk 1 above)
Comment (jpbowen) [reply src/status_im/utils/instabug.cljs:14]: @dmitryn can we reset "Confidence Test Survey ..." to trigger after 5 messages?

Comment (dmitryn) [reply src/status_im/utils/instabug.cljs:14]: @jpbowen yes, we will change it to 5 after "Messaging UX" survey will be finished.

**Comment by chadyj** [comment]
@dmitryn Just got a message from Instabug support
> Apology for taking sometime to get this fixed, Can you please check now and tell me your feedback?
How is it looking?

**Pull request merged**